code/game/machinery/firealarm.dm

```diff
@@ -218,32 +218,32 @@
 
 		switch(buildstage)
 			if(2)
-				if(W.tool_behaviour == TOOL_MULTITOOL)
+				if(tool.tool_behaviour == TOOL_MULTITOOL)
 					detecting = !detecting
 					if (src.detecting)
 						user.visible_message("<span class='notice'>[user] reconnects [src]'s detecting unit!</span>", "<span class='notice'>You reconnect [src]'s detecting unit.</span>")
 					else
 						user.visible_message("<span class='notice'>[user] disconnects [src]'s detecting unit!</span>", "<span class='notice'>You disconnect [src]'s detecting unit.</span>")
 					return
 
-				else if(W.tool_behaviour == TOOL_WIRECUTTER)
+				else if(tool.tool_behaviour == TOOL_WIRECUTTER)
 					buildstage = 1
-					W.play_tool_sound(src)
+					tool.play_tool_sound(src)
 					new /obj/item/stack/cable_coil(user.loc, 5)
 					to_chat(user, "<span class='notice'>You cut the wires from \the [src].</span>")
 					update_appearance()
 					return
 
-				else if(W.force) //hit and turn it on
+				else if(tool.force) //hit and turn it on
 					..()
-					var/area/A = get_area(src)
-					if(!A.fire)
+					var/area/area = get_area(src)
+					if(!area.fire)
 						alarm()
 					return
 
 			if(1)
```

Comment:
if(1)???

Reply:
Why are we using magic numbers 😭

Reply:
cause im scared to touch build stages and dunno what to name them to 😔
i am big brain